contracts/extensions/bridge-operator-governance/BridgeManager.sol

```diff
@@ -0,0 +1,499 @@
+  }
+
+  /**
+   * @inheritdoc IQuorum
+   */
+  function minimumVoteWeight() public view virtual returns (uint256) {
+    return (NUMERATOR_SLOT.mul(TOTAL_WEIGHTS_SLOT.load()) + DENOMINATOR_SLOT.load() - 1) / DENOMINATOR_SLOT.load();
+  }
+
+  /**
+   * @inheritdoc IQuorum
+   */
+  function getThreshold() external view virtual returns (uint256 num_, uint256 denom_) {
+    return (NUMERATOR_SLOT.load(), DENOMINATOR_SLOT.load());
+  }
+
+  /**
+   * @inheritdoc IQuorum
+   */
+  function checkThreshold(uint256 _voteWeight) external view virtual returns (bool) {
+    return _voteWeight * DENOMINATOR_SLOT.load() >= NUMERATOR_SLOT.mul(TOTAL_WEIGHTS_SLOT.load());
+  }
+
+  /**
+   * @dev Internal function to add bridge operators.
+   *
+   * This function adds the specified `bridgeOperators` to the bridge operator set and establishes the associated mappings.
+   *
+   * Requirements:
+   * - The caller must have the necessary permission to add bridge operators.
+   * - The lengths of `voteWeights`, `governors`, and `bridgeOperators` arrays must be equal.
+   *
+   * @param voteWeights An array of uint256 values representing the vote weights for each bridge operator.
+   * @param governors An array of addresses representing the governors for each bridge operator.
+   * @return addeds An array of boolean values indicating whether each bridge operator was successfully added.
+   */
+  function _addBridgeOperators(
+    uint256[] memory voteWeights,
+    address[] memory governors,
+    address[] memory bridgeOperators
+  ) internal nonDuplicate(governors.extend(bridgeOperators)) returns (bool[] memory addeds) {
+    uint256 length = bridgeOperators.length;
+    if (!(length == voteWeights.length && length == governors.length)) revert ErrLengthMismatch(msg.sig);
+    addeds = new bool[](length);
+    // simply skip add operations if inputs are empty.
+    if (length != 0) {
+      EnumerableSet.AddressSet storage _governorSet = _getGovernorsSet();
+      mapping(address => address) storage _governorOf = _getGovernorOf();
+      EnumerableSet.AddressSet storage _bridgeOperatorSet = _getBridgeOperatorSet();
+      mapping(address => BridgeOperatorInfo) storage _governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+
+      // get rid of stack too deep
+      uint256 accumulatedWeight;
+      {
+        address governor;
+        address bridgeOperator;
+        BridgeOperatorInfo memory bridgeOperatorInfo;
+
+        for (uint256 i; i < length; ) {
+          governor = governors[i];
+          bridgeOperator = bridgeOperators[i];
+
+          _requireCreatedEOA(governor);
+          _requireCreatedEOA(bridgeOperator);
+          if (voteWeights[i].toUint96() == 0) revert ErrInvalidVoteWeight(msg.sig);
+
+          addeds[i] = !(_governorSet.contains(governor) &&
+            _governorSet.contains(bridgeOperator) &&
+            _bridgeOperatorSet.contains(governor) &&
+            _bridgeOperatorSet.contains(bridgeOperator));
```

Comment:
```suggestion
          addeds[i] = !(_governorSet.contains(governor) ||
            _governorSet.contains(bridgeOperator) ||
            _bridgeOperatorSet.contains(governor) ||
            _bridgeOperatorSet.contains(bridgeOperator));
```

Should we create another mapping to store all addresses of governors and bridge operators and check in that mapping instead?